support-models/src/main/scala/com/gu/support/redemptions/RedemptionCode.scala

```diff
@@ -0,0 +1,31 @@
+package com.gu.support.redemptions
+
+import com.gu.support.encoding.Codec
+import io.circe.{Decoder, Encoder}
+
+object RedemptionCode {
+  def apply(value: String): Either[String, RedemptionCode] = {
+    // make sure no one can inject anything bad
+    val validChars = List('A' -> 'Z', '0' -> '9', '-' -> '-')
+    val validCharsSet = validChars.flatMap { case (from, to) => (from to to) }.toSet
+    if (value.forall(validCharsSet.contains))
+      Right(new RedemptionCode(value))
+    else {
+      val groupsDesc = validChars.map {
```

Comment:
Is all of this just generating an error message? If so could you move it into a separate method for to keep this one simple